VeganVirus/VeganVirus/FruitThrowAction.cpp

```diff
@@ -0,0 +1,161 @@
+#include "FruitThrowAction.h"
+
+FruitThrowAction::FruitThrowAction(Draw* draw, MouseManager& mouseManager, AudioManager& audioManager) : 
+	Action(FRUIT_THROW_REQ, FRUIT_THROW_ICON), _draw(draw), _mouseManager(mouseManager), _audioManager(audioManager),
+	_mouseDisabledAnimation(draw, FRAME_DURATION, MOUSE_ANIMATION_PATHS, MOUSE_DISABLED_ANIMATION_SIZE, MOUSE_DISABLED_ANIMATION_SIZE)
+{
+	// initialize trail
+	for (int i = 0; i < TRAIL_LENGTH; i++)
+	{
+		_trailPoints.push_back({ 0, 0 });
+	}
+}
+
+FruitThrowAction::~FruitThrowAction()
+{
+	for (FruitThrow* fruit : this->_currentFruits)
```

Comment:
as I mention before I would recommend not using this syntax

Reply:
I see no problem with using for each loops, I think they are clearer than both for loop with an index or for loop with iterators.